### PR TITLE
Fix/dashboard custom range tracking

### LINKED ISSUE #13055: Fix tracking for dashboard performance card and top performers when selecting custom range
Fix tracking for dashboard performance card and top performers when selecting custom range. 

In Android we are tracking some of the custom range dashboard interaction using a legacy event that should've been removed back when we worked on the dynamic dashboard project. The event is `dashboard_stats_custom_range_tab_selected` and was used back when the dashboard had tabs to select the date ranges. 
We track this event every time the user clicks on the drop-down and selectect `custom`. Note the first time the user selects `custom` the event `dashboard_stats_custom_range_add_button_tapped` will be fired. But after adding a custom range subsequent taps on `custom range` option will trigger the legacy event. 

Extra context: pe5sF9-3o5-p2#comment-4155

### PR DESCRIPTION
**Pull request by JorgeMucientes**
Closes: #13055

### Description

In Android we are tracking some of the custom range dashboard interaction using a legacy event that should've been removed back when we worked on the dynamic dashboard project. The event is `dashboard_stats_custom_range_tab_selected` and was used back when the dashboard had tabs to select the date ranges. 
We track this event every time the user clicks on the drop-down and selectect `custom`. Note the first time the user selects `custom` the event `dashboard_stats_custom_range_add_button_tapped` will be fired. But after adding a custom range subsequent taps on `custom range` option will trigger the legacy event. 

Additionally, I renamed the function `onTabSelected` to `onRangeChanged` to improve readability and get rid of the old `tab` concept. 

### Testing information

1. Add `DashboardStatsCard` (aka performance) to the dashboard.
2. Select `custom` range option and set a custom range
3. Tap again on `custom` option from the drop down. 
4. Check that for either step 3 and 4 the event `woocommerceandroid_dashboard_stats_custom_range_add_button_tapped` is tracked.

Repeat the above steps for `TopPerfomersCard` and check tracking is correctly fired. 

### The tests that have been performed

The above. 

### Images/gif

- [x] I have considered if this change warrants release notes and have added them to `RELEASE-NOTES.txt` if necessary. Use the "[Internal]" label for non-user-facing changes.

## Reviewer (or Author, in the case of optional code reviews):

Please make sure these conditions are met before approving the PR, or request changes if the PR needs improvement:

- [x] The PR is small and has a clear, single focus, or a valid explanation is provided in the description. If needed, please request to split it into smaller PRs.
- [x] Ensure Adequate Unit Test Coverage: The changes are reasonably covered by unit tests or an explanation is provided in the PR description.
- [x] Manual Testing: The author listed all the tests they ran, including smoke tests when needed (e.g., for refactorings). The reviewer confirmed that the PR works as expected on big (tablet) and small (phone) in case of UI changes, and no regressions are added.

### CODE REVIEW

**Comment by dangermattic** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Message">
          1 Message
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:book:</td>
      <td data-sticky="false"><p>This PR contains changes to Tracks-related logic. Please ensure (<strong>author and reviewer</strong>) the following are completed:</p>

<ul>
  <li>The tracks events must be validated in the Tracks system.</li>
  <li>Verify the internal Tracks spreadsheet has also been updated.</li>
  <li>Please consider registering any new events.</li>
  <li>The PR must be assigned the <strong>category: tracks</strong> label.</li>
</ul></td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_pr-check">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

File: WooCommerce/src/main/kotlin/com/woocommerce/android/analytics/AnalyticsEvent.kt [hunk 1]
```diff
@@ -167,18 +167,14 @@ enum class AnalyticsEvent(override val siteless: Boolean = false) : IAnalyticsEv
     DASHBOARD_SHARE_YOUR_STORE_BUTTON_TAPPED,
     DASHBOARD_MAIN_STATS_DATE,
     DASHBOARD_MAIN_STATS_LOADED,
-    DASHBOARD_TOP_PERFORMERS_DATE,
     DASHBOARD_TOP_PERFORMERS_LOADED,
-    DASHBOARD_NEW_STATS_REVERTED_BANNER_DISMISS_TAPPED,
-    DASHBOARD_NEW_STATS_REVERTED_BANNER_LEARN_MORE_TAPPED,
```
Comment (JorgeMucientes) [inline WooCommerce/src/main/kotlin/com/woocommerce/android/analytics/AnalyticsEvent.kt:173]: This where simply unused events, so I've removed them

**Review by hafizrahman — APPROVED** [review]
Code looks good and the tracking appear correctly in my test, so I'm approving this.

In my test I also see that the same tracking fires when the "Save" button is tapped here, so in a single flow (tap Custom > date range opened > tap Save), the tracking is triggered twice:



I'm not sure if that's expected or not (seems redundant, or seems like the saving needs a different tracking), but thought I'd mention it.

**Comment by JorgeMucientes** [comment]
Hey @hafizrahman thanks for the review 👍🏼 

> In my test I also see that the same tracking fires when the "Save" button is tapped here, so in a single flow (tap Custom > date range opened > tap Save), the tracking is triggered twice:

Hmm, might be missing something but I can't reproduce that. I don't see the event `stats_custom_range_add_button_tapped` being tracked when tapping on save button.  Moreover I don't see any use of the event `stats_custom_range_add_button_tapped` when tapping on `Save` button from range picker.

**Comment by hafizrahman** [comment]
> Hmm, might be missing something but I can't reproduce that.

Thanks for checking, I restarted AS, cleaned and rebuild my project on this branch and now it doesn't happen again 🤔 Not sure what happened, could be Logcat bugging. I think this should be good to merge now.

**Pull request merged**